### PR TITLE
Create Switch DLS

### PR DESCRIPTION
**Pull request by muhajirdev**
### Summary
TSIA

image

It doesn't really look like the reference (vercel), mainly because of the sizing. (e.g we don't have 14px font size token)

API and design is derived from https://vercel.com/design/switch

### CODE REVIEW

File: src/components/dls/Switch/Switch.tsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+type Item = {
+  name: string;
+  value: string;
+  width: number;
+  disabled?: boolean;
+};
+export enum SwitchSize {
+  Small = 'small',
+  Normal = 'normal',
```
Comment (osamasayed) [inline src/components/dls/Switch/Switch.tsx:15]: I think "Normal" should be renamed to "Medium" and also inside the css class names for the sake of consistency with other components when it comes to sizing

File: src/components/dls/Switch/Switch.tsx [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          disabled={item.disabled}
+          type="button"
+          className={classNames(styles.item, selected === item.value && styles.itemSelected, {
+            [styles.itemLarge]: size === SwitchSize.Large,
+            [styles.itemNormal]: size === SwitchSize.Normal,
+            [styles.itemSmall]: size === SwitchSize.Small,
+          })}
+          key={item.value}
+          onClick={() => onSelect(item.value)}
+          style={{ width: item.width }}
```
Comment (osamasayed) [inline src/components/dls/Switch/Switch.tsx:39]: I don't think we should have a flexible width value here. The size is what should control the width

File: src/components/dls/Switch/Switch.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
+import classNames from 'classnames';
+
+import styles from './Switch.module.scss';
+
+// reference: https://vercel.com/design/switch
+
+type Item = {
+  name: string;
+  value: string;
+  width: number;
```
Comment (osamasayed) [inline src/components/dls/Switch/Switch.tsx:10]: we shouldn't use `width`

File: src/components/dls/Switch/Switch.tsx (hunk 1 above)
Comment (muhajirdev) [reply src/components/dls/Switch/Switch.tsx:15]: agree 👍

Comment (tifa2UP) [reply src/components/dls/Switch/Switch.tsx:15]: ditto

File: src/components/dls/Switch/Switch.tsx (hunk 3 above)
Comment (tifa2UP) [reply src/components/dls/Switch/Switch.tsx:10]: yeah, let's not make width a prop. We can have a `fullWidth` option that takes 100% of the container width.

**Review by tifa2UP — APPROVED** [review]

**Review by tifa2UP — APPROVED** [review]

File: src/components/dls/Switch/Switch.module.scss [hunk 4]
```diff
@@ -0,0 +1,45 @@
+.container {
```
Comment (tifa2UP) [inline src/components/dls/Switch/Switch.module.scss:1]: switch doesn't appear to be working on dark mode

**Pull request merged**